mono/utils/mono-threads.h

```diff
@@ -464,7 +464,7 @@ mono_thread_info_describe_interrupt_token (THREAD_INFO_TYPE *info, GString *text
 gboolean
 mono_thread_info_is_live (THREAD_INFO_TYPE *info);
 
-HANDLE
+MonoThreadInfo*
 mono_threads_create_thread (MonoThreadStart start, gpointer arg, MonoThreadParm *tp, MonoNativeThreadId *out_tid);
 
```

Comment:
The handle also has the same lifetime as the `MonoThreadInfo`, so the only difference is how we might crash. If we want to fix that on the longer term, we would have to do it another way anyway.